plugins/out_s3/s3.c

```diff
@@ -1332,24 +1357,9 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
     flb_sds_destroy(s3_key);
     uri = tmp;
 
```

Comment:
so in this way, `s3_put_object` will not compress data and I notice that here also uses this function: https://github.com/fluent/fluent-bit/blob/d3ff30847842867b4d7a7aa5eede5e7a390b2637/plugins/out_s3/s3.c#L1216. I didn't dive deep but What if we need to compress data in line 1216?

Reply:
So put_all_chunks will upload all part we store them, and in this code, we will compress data before store them, So it should be good. And in the testing, I have purpose break the upload to test, all good

Reply:
I think Zhonghui is correct, compression needs to happen in the `put_all_chunks` case as well. 

> we will compress data before store them

This is not the case in the existing code and I do not see this logic in the new code either. IIRC, yesterday we chatted and agreed that special compression code needed to be added in `put_all_chunks`

Reply:
Will do